blik/src/main/java/com/adyen/checkout/blik/BlikView.kt

```diff
@@ -0,0 +1,108 @@
+import com.adyen.checkout.components.ui.view.AdyenTextInputEditText
+import com.adyen.checkout.core.exception.CheckoutException
+import com.adyen.checkout.core.log.LogUtil.getTag
+import com.adyen.checkout.core.log.Logger
+import com.google.android.material.textfield.TextInputLayout
+
+class BlikView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
+    AdyenLinearLayout<BlikOutputData, BlikConfiguration, PaymentComponentState<BlikPaymentMethod>, BlikComponent>(context, attrs, defStyleAttr),
+    Observer<BlikOutputData> {
+
+    private var blikInputData = BlikInputData()
+    private lateinit var blikCodeInput: TextInputLayout
+    private lateinit var blikCodeEditText: AdyenTextInputEditText
+
+    init {
+        orientation = VERTICAL
+        LayoutInflater.from(getContext()).inflate(R.layout.blik_view, this, true)
+        val padding = resources.getDimension(R.dimen.standard_margin).toInt()
+        setPadding(padding, padding, padding, 0)
+    }
+
+    override fun initLocalizedStrings(localizedContext: Context) {
+        val myAttrs = intArrayOf(android.R.attr.hint)
+        val typedArray = localizedContext.obtainStyledAttributes(R.style.AdyenCheckout_Blik_BlikCodeInput, myAttrs)
+        blikCodeInput.hint = typedArray.getString(0)
+        typedArray.recycle()
+    }
+
+    override fun initView() {
+        blikCodeInput = findViewById(R.id.textInputLayout_blikCode) ?: throw CheckoutException("Could not find views inside layout.")
+        blikCodeEditText = blikCodeInput.editText as AdyenTextInputEditText
+
+        blikCodeEditText.setOnChangeListener {
+            blikInputData.blikCode = blikCodeEditText.rawValue
+            notifyInputDataChanged()
+            blikCodeInput.error = null
+        }
+
+        blikCodeEditText.onFocusChangeListener = OnFocusChangeListener { _: View?, hasFocus: Boolean ->
+            val outputData = component.outputData
+            val blikCodeValidation = outputData?.blikCodeField?.validation
+            if (hasFocus) {
+                blikCodeInput.error = null
+            } else if (blikCodeValidation != null && !blikCodeValidation.isValid()) {
+                val errorReasonResId = (blikCodeValidation as Validation.Invalid).reason
+                blikCodeInput.error = localizedContext.getString(errorReasonResId)
+            }
+        }
+    }
+
+    override fun onChanged(blikOutputData: BlikOutputData?) {
+        Logger.v(TAG, "blikOutputData changed")
+    }
+
+    override fun onComponentAttached() {
+        // nothing to impl
+    }
+
+    override fun observeComponentChanges(lifecycleOwner: LifecycleOwner) {
+        component.observeOutputData(lifecycleOwner, this)
+    }
+
+    override val isConfirmationRequired: Boolean
+        get() = true
+
+    override fun highlightValidationErrors() {
+        Logger.d(TAG, "highlightValidationErrors")
+        val outputData: BlikOutputData? = if (component.outputData != null) {
```

Comment:
`val outputData = component.outputData ?: return` is shorter and also removes the need for the `!!` below

Reply:
Yep, missed this one, thx!